src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -292,5 +306,72 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            if (!TryReadFile(statusFilePath, reusableReader, out string statusFileContents))
+            {
+                // Between the time that we get an ID and the time that we try to read the associated stat
+                // file(s), the process could be gone.
+                result = default(ParsedStatus);
+                return false;
+            }
+
+            var parser = new StringParser(statusFileContents, '\n');
+            var results = default(ParsedStatus);
+
+            while (parser.MoveNext())
+            {
+                string[] elements = parser.ExtractCurrent().Split(Delimiters, 3, StringSplitOptions.RemoveEmptyEntries);
+                if (elements.Length < 2)
+                {
+                    continue;
+                }
+
+                switch (elements[0])
+                {
+                    case "Pid":
+                        results.Pid = int.Parse(elements[1]);
+                        break;
+                    case "VmHWM":
+                        results.VmHWM = ulong.Parse(elements[1]) * 1024;
+                        break;
+                    case "VmRSS":
+                        results.VmRSS = ulong.Parse(elements[1]) * 1024;
+                        break;
+                    case "VmData":
+                        results.VmData = ulong.Parse(elements[1]) * 1024;
+                        break;
+                    case "VmSwap":
+                        results.VmSwap = ulong.Parse(elements[1]) * 1024;
+                        break;
+                    case "VmSize":
+                        results.VmSize = ulong.Parse(elements[1]) * 1024;
+                        break;
+                    case "VmPeak":
+                        results.VmPeak = ulong.Parse(elements[1]) * 1024;
+                        break;
+                }
+            }
+            result = results;
+            return true;
+        }
+
+        private static bool TryReadFile(string filePath, ReusableTextReader reusableReader, out string fileContents)
+        {
+            try
+            {
+                using (var fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false))
```

Comment:
Out of curiosity: why `bufferSize: 1, useAsync: false`?

Reply:
this is just moving logic so don't expecting the answer I guess

Reply:
Yep, just moved logic here.

Maybe the original commit can help: https://github.com/dotnet/corefx/commit/12a2c2afed6814cf3a0862046f84b40da4fb1d67

Reply:
useAsync:false because we're doing synchronous access.

bufferSize:1 to avoid allocating an unnecessary buffer inside of FileStream.